api/tokens/views.py

```diff
@@ -0,0 +1,98 @@
+"""
+Views related to person access tokens. Intended for OSF internal use only
+"""
+from rest_framework.exceptions import APIException
+from rest_framework import generics
+from rest_framework import renderers
+from rest_framework import permissions as drf_permissions
+
+from modularodm import Q
+
+from framework.auth import cas
+from framework.auth.oauth_scopes import ComposedScopes
+
+from api.base.filters import ODMFilterMixin
+from api.base.utils import get_object_or_error
+from api.base import permissions as base_permissions
+from api.tokens.serializers import ApiOAuth2PersonalTokenSerializer
+
+from website.models import ApiOAuth2PersonalToken
+
+
+class TokenList(generics.ListCreateAPIView, ODMFilterMixin):
+    """
+    Get a list of personal access tokens that the user has registered
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        base_permissions.OwnerOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [ComposedScopes.FULL_READ]
+    required_write_scopes = [ComposedScopes.FULL_WRITE]
+
+    serializer_class = ApiOAuth2PersonalTokenSerializer
+
+    renderer_classes = [renderers.JSONRenderer]  # Hide from web-browsable API tool
+
+    def get_default_odm_query(self):
+
+        user_id = self.request.user._id
+        return (
+            Q('owner', 'eq', user_id) &
+            Q('is_active', 'eq', True)
+        )
+
+    # overrides ListAPIView
+    def get_queryset(self):
+        query = self.get_query_from_request()
+        return ApiOAuth2PersonalToken.find(query)
+
+    def perform_create(self, serializer):
+        """Add user to the created object"""
+        serializer.validated_data['owner'] = self.request.user
+        serializer.save()
+
+
+class TokenDetail(generics.RetrieveUpdateDestroyAPIView):
+    """
+    Get information about a specific personal access token that the user has registered
+
+    Should not return information if the token belongs to a different user
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        base_permissions.OwnerOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [ComposedScopes.FULL_READ]
```

Comment:
See note about scopes for read and write field above.

If the list and detail views serialize the same basic data, you can reuse the same CoreScope, but they should be unique to tokens.